aws/src/main/java/org/apache/iceberg/aws/glue/LockManagers.java

```diff
@@ -0,0 +1,240 @@
+    } catch (NoSuchMethodException e) {
+      throw new IllegalArgumentException(String.format(
+          "Cannot initialize LockManager, missing no-arg constructor: %s", impl), e);
+    }
+
+    LockManager lockManager;
+    try {
+      lockManager = ctor.newInstance();
+    } catch (ClassCastException e) {
+      throw new IllegalArgumentException(
+          String.format("Cannot initialize LockManager, %s does not implement LockManager.", impl), e);
+    }
+
+    lockManager.initialize(properties);
+    return lockManager;
+  }
+
+  abstract static class LockManagerBase implements LockManager {
+
+    private static volatile ScheduledExecutorService scheduler;
+
+    private long acquireTimeoutMs;
+    private long acquireIntervalMs;
+    private long heartbeatIntervalMs;
+    private long heartbeatTimeoutMs;
+    private int heartbeatThreads;
+
+    public long heartbeatTimeoutMs() {
+      return heartbeatTimeoutMs;
+    }
+
+    public long heartbeatIntervalMs() {
+      return heartbeatIntervalMs;
+    }
+
+    public long acquireIntervalMs() {
+      return acquireIntervalMs;
+    }
+
+    public long acquireTimeoutMs() {
+      return acquireTimeoutMs;
+    }
+
+    public int heartbeatThreads() {
+      return heartbeatThreads;
+    }
+
+    @SuppressWarnings("StaticGuardedByInstance")
+    public ScheduledExecutorService scheduler() {
+      if (scheduler == null) {
+        synchronized (this) {
+          if (scheduler == null) {
+            scheduler = Executors.newScheduledThreadPool(heartbeatThreads);
+          }
+        }
+      }
+      return scheduler;
+    }
+
+    @Override
+    public void initialize(Map<String, String> properties) {
+      this.acquireTimeoutMs = PropertyUtil.propertyAsLong(properties,
+          CatalogProperties.LOCK_ACQUIRE_TIMEOUT_MS, CatalogProperties.LOCK_ACQUIRE_TIMEOUT_MS_DEFAULT);
+      this.acquireIntervalMs = PropertyUtil.propertyAsLong(properties,
+          CatalogProperties.LOCK_ACQUIRE_INTERVAL_MS, CatalogProperties.LOCK_ACQUIRE_INTERVAL_MS_DEFAULT);
+      this.heartbeatIntervalMs = PropertyUtil.propertyAsLong(properties,
+          CatalogProperties.LOCK_HEARTBEAT_INTERVAL_MS, CatalogProperties.LOCK_HEARTBEAT_INTERVAL_MS_DEFAULT);
+      this.heartbeatTimeoutMs = PropertyUtil.propertyAsLong(properties,
+          CatalogProperties.LOCK_HEARTBEAT_TIMEOUT_MS, CatalogProperties.LOCK_HEARTBEAT_TIMEOUT_MS_DEFAULT);
+      this.heartbeatThreads = PropertyUtil.propertyAsInt(properties,
+          CatalogProperties.LOCK_HEARTBEAT_THREADS, CatalogProperties.LOCK_HEARTBEAT_THREADS_DEFAULT);
+    }
+  }
+
+  /**
+   * Default implementation of {@link LockManager} that uses an in-memory concurrent map for locking.
+   * This implementation should only be used for testing,
+   * or if the caller only needs locking within the same JVM during table commits.
+   */
+  static class DefaultLockManager extends LockManagerBase {
```

Comment:
Could you rename this to `InMemoryLockManager`? I think we should make it clear that this is not suitable for real world use. This should probably also be in test so that people don't try to use it at runtime?

Reply:
Will do